sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -510,11 +529,10 @@ def track_user_pipelines(self):
     function also clears up internal states for those anonymous pipelines once
     all their PCollections are anonymous.
     """
-    self._tracked_user_pipelines = set()
```

Comment:
With this change, what if a user pipeline is no longer in scope, will it get cleared from the `self._tracked_user_pipelines`?